crates/uv-keyring/src/lib.rs

```diff
@@ -0,0 +1,608 @@
+    #[cfg(all(target_os = "macos", feature = "apple-native"))]
+    return macos::default_credential_builder();
+    #[cfg(all(target_os = "windows", feature = "windows-native"))]
+    return windows::default_credential_builder();
+    #[cfg(not(any(
+        all(target_os = "linux", feature = "secret-service"),
+        all(target_os = "freebsd", feature = "secret-service"),
+        all(target_os = "openbsd", feature = "secret-service"),
+        all(target_os = "macos", feature = "apple-native"),
+        all(target_os = "windows", feature = "windows-native"),
+    )))]
+    credential::nop_credential_builder()
+}
+
+fn build_default_credential(target: Option<&str>, service: &str, user: &str) -> Result<Entry> {
+    static DEFAULT: std::sync::LazyLock<Box<CredentialBuilder>> =
+        std::sync::LazyLock::new(default_credential_builder);
+    let guard = DEFAULT_BUILDER
+        .read()
+        .expect("Poisoned RwLock in keyring-rs: please report a bug!");
+    let builder = guard.inner.as_ref().unwrap_or_else(|| &DEFAULT);
+    let credential = builder.build(target, service, user)?;
+    Ok(Entry { inner: credential })
+}
+
+#[derive(Debug)]
+pub struct Entry {
+    inner: Box<Credential>,
+}
+
+impl Entry {
+    /// Create an entry for the given service and user.
+    ///
+    /// The default credential builder is used.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an [`Error`] if the `service` or `user` values are invalid.
+    /// The specific reasons for invalidity are platform-dependent, but include length constraints.
+    ///
+    /// # Panics
+    ///
+    /// In the very unlikely event that the internal credential builder's `RwLock` is poisoned, this function
+    /// will panic. If you encounter this, and especially if you can reproduce it, please report a bug with the
+    /// details (and preferably a backtrace) so the developers can investigate.
+    pub fn new(service: &str, user: &str) -> Result<Self> {
+        let entry = build_default_credential(None, service, user)?;
+        Ok(entry)
+    }
+
+    /// Create an entry for the given target, service, and user.
+    ///
+    /// The default credential builder is used.
+    pub fn new_with_target(target: &str, service: &str, user: &str) -> Result<Self> {
+        let entry = build_default_credential(Some(target), service, user)?;
+        Ok(entry)
+    }
+
+    /// Create an entry from a credential that may be in any credential store.
+    pub fn new_with_credential(credential: Box<Credential>) -> Self {
+        Self { inner: credential }
+    }
+
+    /// Set the password for this entry.
+    ///
+    /// Can return an [`Ambiguous`](Error::Ambiguous) error
+    /// if there is more than one platform credential
+    /// that matches this entry.  This can only happen
+    /// on some platforms, and then only if a third-party
+    /// application wrote the ambiguous credential.
+    pub async fn set_password(&self, password: &str) -> Result<()> {
```

Comment:
`async` methods on `Entry` have been updated from the sync versions in `keyring-rs`.